DOCSP-9596 replace one

ccho-mongodb opened the pull request:
## Pull Request Info

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-9596

### Docs staging link (requires sign-in on MongoDB Corp SSO):
[replaceOne page](https://docs-mongodbcom-staging.corp.mongodb.com/e0d5956/java/docsworker-xlarge/DOCSP-9596-replace-one/usage-examples/replaceOne)

[updateOne update](https://docs-mongodbcom-staging.corp.mongodb.com/e0d5956/java/docsworker-xlarge/DOCSP-9596-replace-one/usage-examples/updateOne)

Note: the API links are currently broken as the 4.x docs appear to have been moved. Need to update the snooty toml once I confirm.

nathan-contino-mongo commented on source/usage-examples/replaceOne.txt:
```diff
@@ -0,0 +1,110 @@
+
+Upon successful execution, the ``replaceOne()`` method returns an instance
+of ``UpdateResult``.  You can retrieve information such as the number of
+documents modified calling the ``getModifiedCount()`` method. You can also
```
```suggestion
documents modified by calling the ``getModifiedCount()`` method. You can also
```

I think you accidentally a word here!

nathan-contino-mongo commented on source/usage-examples/replaceOne.txt:
```diff
@@ -0,0 +1,110 @@
+documents modified calling the ``getModifiedCount()`` method. You can also
+retrieve the value of the document's ``_id`` field by calling the
+``getUpsertedId()`` method if you set ``upsert(true)`` in the
+``UpdateOptions`` instance and the operation resulted in an upsert.
```
```suggestion
``UpdateOptions`` instance and the operation resulted in the insertion of a new document.
```

I would say that the operation is an upsert no matter what, but can result in either an update of an existing document or an insert of a new document.

nathan-contino-mongo commented on source/usage-examples/replaceOne.txt:
```diff
@@ -0,0 +1,110 @@
+
+The following snippet uses the following objects and methods:
+
+- A **query filter** that is passed in the ``replaceOne()`` method. The ``eq``
```
This could be a good opportunity to link the the manual's entry for ``eq``. That being said, if you think that's too much linking that's fine too.

ccho-mongodb replied:
It's linked at the bottom; I know this is not how our team initially approached driver documentation, but this is in response to feedback about having links too early in the document.

jyemin commented on source/usage-examples/replaceOne.txt:
```diff
@@ -0,0 +1,110 @@
+.. literalinclude:: /includes/usage-examples/code-snippets/ReplaceOne.java
+   :language: java
+
+If you are using the **legacy API**, see (TODO: FAQ ?) to determine what
```
Resolve TODO before pushing?

ccho-mongodb replied:
I'm using TODOs only as placeholders. I'll start creating the placeholders soon to avoid having to use these.

jyemin commented on source/includes/usage-examples/code-snippets/ReplaceOne.java:
```diff
@@ -0,0 +1,47 @@
+        MongoDatabase database = mongoClient.getDatabase("sample_mflix");
+        MongoCollection<Document> collection = database.getCollection("movies");
+
+        try {
```
Can't remember if we discussed this already, but should we use try-with-resources here e.g.

    try(MongoClient mongoClient = MongoClients.create(uri))
         MongoDatabase database = mongoClient.getDatabase("sample_mflix");
         MongoCollection<Document> collection = database.getCollection("movies");
         ...

jyemin replied:
It's more modern Java

ccho-mongodb replied:
Thanks, will update. I now recall we discussed try-with-resources in the context of opening a file for the CSFLE guide. Will fix for this PR and revisit other code examples in docs-java.

ccho-mongodb commented:
Revisited the other usage examples to use try-with-resources. Also removed a couple "TODO" links as they should be at the bottom anyway, and added some upsert-specific output in the relevant examples.

rozza commented on source/includes/usage-examples/code-snippets/InsertOne.java:
```diff
@@ -16,21 +17,27 @@
-            System.err.println("Unable to insert due to an error: " + me);
+                System.out.println("Success! Inserted document id: " + result.getInsertedId());
+
+                // Retrieve the inserted document
```
Is this comment staying or a todo?

ccho-mongodb replied:
Whoops, will remove.

rozza commented:
LGTM
